cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -0,0 +1,122 @@
+package relayer
+
+// -----------------------------------------------------
+//      Ethereum relayer
+//
+//      Initializes the relayer service, which parses,
+//      encodes, and packages named events on an Ethereum
+//      Smart Contract for validator's to sign and send
+//      to the Cosmos bridge.
+// -----------------------------------------------------
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	amino "github.com/tendermint/go-amino"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/cosmos/peggy/cmd/ebrelayer/contract"
+	"github.com/cosmos/peggy/cmd/ebrelayer/events"
+	"github.com/cosmos/peggy/cmd/ebrelayer/txs"
+)
+
+// InitEthereumRelayer : Starts an event listener on a specific Ethereum network, contract, and event
+func InitEthereumRelayer(cdc *amino.Codec, chainID string, provider string, contractAddress common.Address, cosmosSupport bool, validatorName string, passphrase string, validatorAddress sdk.ValAddress) error {
+	// Start client with infura ropsten provider
+	client, err := SetupWebsocketEthClient(provider)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("\nStarted ethereum websocket with provider: %s", provider)
+
+	// Load our contract's ABI
+	contractABI := contract.LoadABI(cosmosSupport)
+
+	var targetContract txs.ContractRegistry
+	var eventName string
+	var eventSignature string
+
+	switch cosmosSupport {
+	case true:
+		targetContract = txs.CosmosBridge
+		eventName = "LogNewProphecyClaim"
+		eventSignature = contractABI.Events["LogNewProphecyClaim"].Id().String()
+	case false:
+		targetContract = txs.BridgeBank
+		eventName = "LogLock"
+		eventSignature = contractABI.Events["LogLock"].Id().Hex()
+	}
+
+	// Get the specific contract's address (CosmosBridge or BridgeBank)
+	targetAddress, err := txs.GetAddressFromBridgeRegistry(client, contractAddress, targetContract)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// We need the target address in bytes[] for the query
+	query := ethereum.FilterQuery{
+		Addresses: []common.Address{targetAddress},
+	}
+
+	// We will check logs for new events
+	logs := make(chan types.Log)
+
+	// Filter by contract and event, write results to logs
+	sub, err := client.SubscribeFilterLogs(context.Background(), query, logs)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("\nSubscribed to %v contract at address: %s\n", targetContract, targetAddress.Hex())
```

Comment:
can we standardize the commands to use a logger or plain print statements? I prefer the former if you ask me

Reply:
What would be the ideal logger solution? go-ethereum has https://github.com/ethereum/go-ethereum/tree/master/log, but right now `InitCosmosRelayer()` uses a tendermint logger from github.com/tendermint/tendermint/libs/log. I favor a unified logging solution so we don't duplicate work if there's a future PR to combine the cosmos/ethereum relayer (I've created an issue for this https://github.com/cosmos/peggy/issues/88).

Reply:
Would it make sense for this to wait until the issue is addressed or should we try to implement something as part of this PR?